tests/configs/basic_app/kubetools.yml

```diff
@@ -25,3 +25,12 @@ deployments:
           timeoutSeconds: 5
           httpGet:
             path: /ping
+
+cronjobs:
+  generic-cronjob:
+    schedule: "*/1 * * * *"
+    concurrency_policy: "Allow"
+    containers:
+      generic-container:
+        image: generic-image
+        command: [generic-command]
```

Comment:
This would be more in line with the spirit of this test case
```suggestion
    containers:
      generic-container:
        containerContext: generic-context
```

Reply:
Done, here is the commit link https://github.com/EDITD/kubetools/pull/118/commits/abf557fb47b73c192123c2667795be92cbe6e4d7

Reply:
![image](https://user-images.githubusercontent.com/8948373/151381799-287bda50-e538-478e-9779-9a5f71e57324.png)
Typical example where a functional code change has been added in an unrelated commit

Reply:
For this change:
```
    'upgrades',
     'deployments',
     'dependencies',
     'cronjobs',
```

I added the `cronjobs` object because otherwise it doesn't link the `containerContext` inside the `cronjobs` object to the `containerContext` that is inside the list of `containerContext`

So it's link to the commit in the sens of, if I don't add this small change the `pytest --cov -v` wouldn't work

Reply:
>  I added the cronjobs object because otherwise it doesn't link the containerContext inside the cronjobs object to the containerContext that is inside the list of containerContext

Ok. Makes sense. The structure of the code is quite misleading, because it seems that the logic for "conditions" is driving the use of containerContext... this is quite confusing

Reply:
This conversation is :+1: for me. Will keep it open for this explanation^^